### PR TITLE
 Deprecate the AES_ige_*() functions (take 3)

### PR DESCRIPTION
**Pull request by mattcaswell**
These functions were already partially deprecated. Now we do it fully.

This is an alternative approach to that shown in #10540, and also to the one in #10538 and #10367.

### CODE REVIEW

**Comment by levitte** [comment]
This makes sense to me.

File: include/openssl/aes.h [hunk 1]
```diff
@@ -73,17 +73,16 @@ void AES_cfb8_encrypt(const unsigned char *in, unsigned char *out,
...
 /* NB: the IV is _four_ blocks long */
-void AES_bi_ige_encrypt(const unsigned char *in, unsigned char *out,
-                        size_t length, const AES_KEY *key,
-                        const AES_KEY *key2, const unsigned char *ivec,
-                        const int enc);
-# endif
+DEPRECATEDIN_3_0(void AES_bi_ige_encrypt(
+    const unsigned char *in, unsigned char *out, size_t length,
+    const AES_KEY *key, const AES_KEY *key2, const unsigned char *ivec,
+    const int enc))
```
Comment (levitte) [inline include/openssl/aes.h]: Is this alignment according to code style?  I'm sure @DDvO's checker script is gonna say a few things about these lines.

(and yes, I'm perfectly aware how very ugly the alternative is.  Still...)

**Comment by t8m** [comment]
This has a disadvantage that you cannot find out where openssl itself uses the deprecated APIs with the simple addition of -Wdeprecated-declarations flag to the config.

**Comment by mattcaswell** [comment]
> This has a disadvantage that you cannot find out where openssl itself uses the deprecated APIs with the simple addition of -Wdeprecated-declarations flag to the config.

True, although I don't anticipate that we would need to do that very often. Where we do I imagine it is only us (i.e. the committer team) that would need to, and the deprecations can be re-enabled with a very simple edit to macros.h

**Comment by levitte** [comment]
no-deprecated should still be useful in that regard

File: include/openssl/aes.h (hunk 1 above)
Comment (DDvO) [reply include/openssl/aes.h]: > Is this alignment according to code style? I'm sure @DDvO's checker script is gonna say a few things about these lines.
> 
> (and yes, I'm perfectly aware how very ugly the alternative is. Still...)

Indeed, `check-format.pl` complains on that decl and the one before like this:
```
aes.h:79:indent = 4 != 38 for hanging '(':    unsigned char *out, size_t length, const AES_KEY *key, unsigned char *ivec,
aes.h:80:indent = 4 != 38 for hanging '(':    const int enc))
aes.h:83:indent = 4 != 41 for hanging '(':    const unsigned char *in, unsigned char *out, size_t length,
aes.h:84:indent = 4 != 41 for hanging '(':    const AES_KEY *key, const AES_KEY *key2, const unsigned char *ivec,
aes.h:85:indent = 4 != 41 for hanging '(':    const int enc))
```

A viable alternative formatting could be:
```
/* NB: the IV is _two_ blocks long */
DEPRECATEDIN_3_0(void
                 AES_ige_encrypt(const unsigned char *in, unsigned char *out,
                                 size_t length, const AES_KEY *key,
                                 unsigned char *ivec, const int enc))
/* NB: the IV is _four_ blocks long */
DEPRECATEDIN_3_0(void
                 AES_bi_ige_encrypt(const unsigned char *in, unsigned char *out,
                                    size_t length, const AES_KEY *key,
                                    const AES_KEY *key2,
                                    const unsigned char *ivec, const int enc))
```

Comment (richsalz) [reply include/openssl/aes.h]: Does this work?
```
DEPRECATEDIN_3_0(
    void
    AES_bi_ige_encrypt(const unsigned char *in, unsigned char *out, size_t length,
                       const AES_KEY *key, const AES_KEY *key2,
                       const unsigned char *ivec, const int enc)
)
```
And ```const int``` is pointless BTW.

Comment (DDvO) [reply include/openssl/aes.h]: The tool does not like that, neither does the Emacs C mode.

I've already added the possibility for such a relaxation in the tool the allows for alternative indentation like this within expressions, but not (yet?) at statement/declaration level.

**Comment by mattcaswell** [comment]
I updated the formatting in line with the suggestion from @DDvO.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**